### PR TITLE
update GCC_ARM compiler supported to 10.3-2021.07

### PR DESCRIPTION
**Pull request by saheerb**

### CODE REVIEW

File: docs/tools/tools_intro.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ You can build Mbed OS with the Arm Compiler and GNU Arm Embedded toolchains. The
 | Compiler| Download location | Name in Mbed CLI |
 | --- | --- | --- |
 | Arm Compiler 6.16 (default ARM toolchain) | - A paid version is available as [Arm Compiler 6.16 Professional](https://developer.arm.com/products/software-development-tools/compilers/arm-compiler/downloads/version-6). </br> - A paid version is also included in [Keil MDK 5.34](https://www.keil.com/update/relnotes/MDK534.htm) | `ARM` |
-| GNU Arm Embedded version 9 (9-2019-q4-major) | [GNU Arm Embedded version 9 (9-2019-q4-major)](https://developer.arm.com/open-source/gnu-toolchain/gnu-rm/downloads) | `GCC_ARM` |
+| GNU Arm Embedded version 10 (9-2019-q4-major) | [GNU Arm Embedded version 10 (10.3-2021.07)](https://developer.arm.com/open-source/gnu-toolchain/gnu-rm/downloads) | `GCC_ARM` |
```
Comment (donatieng) [inline docs/tools/tools_intro.md]:
```suggestion
| GNU Arm Embedded version 10 (10.3-2021.07) | [GNU Arm Embedded version 10 (10.3-2021.07)](https://developer.arm.com/open-source/gnu-toolchain/gnu-rm/downloads) | `GCC_ARM` |
```

**Review by donatieng — CHANGES_REQUESTED** [review]
Thanks Saheer, one minor nit below:

**Comment by jeromecoutant** [comment]
Need to wait for updates in mbed-os repo ?

**Comment by donatieng** [comment]
Indeed, prior to merging we need:
* https://github.com/ARMmbed/mbed-os/issues/14968
* https://github.com/ARMmbed/mbed-os/pull/15007

**Review by donatieng — APPROVED** [review]
@ChrisGooch-Arm let's wait for the above issue and PR to be in first, they we can merge this one

**Review by 0xc0170 — APPROVED** [review]

**Comment by 0xc0170** [comment]
Both of above will be most likely resolved within today, I'll keep an eye on both.

**Comment by ChrisGooch-Arm** [comment]
@donatieng, @0xc0170  - is this ok to merge now or is there still a dependency on other things?

**Comment by 0xc0170** [comment]
Yes, please lets merge. All was resolved for the Mbed OS release in September.

**Comment by 0xc0170** [comment]
@ChrisGooch-Arm can we merge this, its needed.

**Pull request merged**